ocaml/driver/makedepend.ml

```diff
@@ -327,11 +328,17 @@ let read_parse_and_extract parse_function extract_function def ast_kind
     end
   with x -> begin
     print_exception x;
-    if not !allow_approximation then begin
+    if !strict then (
       Error_occurred.set ();
       (String.Set.empty, def)
-    end else
-      (read_and_approximate source_file, def)
+    )
+    else (
+      if not !allow_approximation then begin
+        Error_occurred.set ();
+        (String.Set.empty, def)
```

Comment:
Seems like this can be merged with the above branch?

Reply:
Fixed